### ISSUE TITLE
Filter weights in Locally connected layer

### ISSUE THREAD

**Issue by rohith513** [keras-team/keras#13124]
**System information**  
- Have I written custom code (as opposed to using example directory):  No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Arch Linux
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.13.1
- Keras version:  2.1.6
- Python version:  3.7
- CUDA/cuDNN version:  10
- GPU model and memory:  Nvidia GTX 1070

I have a problem understanding how the filters are used in Locally connected layer.
For example, say input is 6x6x3 image and we use one Conv2D(same padding) and one LocallyConnected2D with 4 filters each of size 3x3. 

`filters, biases = model.layers[2].get_weights()`

When I use layer.get_weights() on conv2D, it returns filters with shape (3,3,3,4) and bias shape  (4, ) which is expected as we have 4 filters with shape 3x3x3.
But layer.get_weights() on LocallyConnected2D returns filters with shape (16,36,4) and bias shape (4,4,4).
Why is the filter shape 16x36?
I know that locally connected layer uses different filters at each input patch. How do we slide across the whole image with only 4 filters?

**Comment by rohith513** [comment]
@joelthchao Can you please look in to this?

**Comment by rohith513** [comment]
Can someone explain how the filters are slided across the image here when only few filters are used?s
Are filters are reused like in Tiled convolutions?
@fchollet

**Comment by joelthchao** [comment]
Similar to conv2d. However, when you slide along x and y dimension, no filter is reused in different positions.
